src/Language/PureScript/TypeChecker/Monad.hs

```diff
@@ -75,11 +75,16 @@ data CheckState = CheckState
   -- This goes into state, rather than using 'rethrow',
   -- since this way, we can provide good error messages
   -- during instance resolution.
+  , checkDesugaringWarnings :: MultipleErrors
```

Comment:
Looks pretty hacky to me! Maybe this is the right thing to do when all things are considered, but the alternative that comes to mind is writing the set of constructors that are used by the solver, and moving import linting out of import desugaring and into its own step that runs after type checking and uses that set. Is that a bad idea for some reason I'm not seeing? Or just not worth doing?

Reply:
I thought I wouldn’t be able to lint imports after typechecking so easily, hence the minimal hack, but this isn’t that bad after all!